crates/re_arrow_store/src/store_helpers.rs

```diff
@@ -7,22 +7,22 @@ use crate::{DataStore, LatestAtQuery};
 // --- Read ---
 
 impl DataStore {
-    /// Get the latest value for a given [`re_types::Component`].
+    /// Get the latest value for a given [`re_types::Component`] and the associated [`RowId`].
     ///
     /// This assumes that the row we get from the store only contains a single instance for this
     /// component; it will log a warning otherwise.
     ///
     /// This should only be used for "mono-components" such as `Transform` and `Tensor`.
     ///
     /// This is a best-effort helper, it will merely log errors on failure.
-    pub fn query_latest_component<C: Component>(
+    pub fn query_latest_component_and_row_id<C: Component>(
```

Comment:
Rather than "component_and_row_id" what if this the result-type was just a "VersionedComponent"?
```
pub fn query_latest_component<C: Component>(
        &self,
        entity_path: &EntityPath,
        query: &LatestAtQuery,
    ) -> Option<VersionedComponent<C>>
```

Reply:
Interesting. I'll try and see how this looks.

Reply:
I've implemented it and I think it's a nice enough improvement as it at least prevents the combinatorial explosion of `query_latest_*` APIs.

Overall the whole thing is definitely not _great_, there is so much more we could do if the notion of versioning was more tightly integrated (or rather: integrated at all) all over the codebase overall but... gotta :ship: !